internal/worker/storageprovisioner/volume_ops.go

```diff
@@ -170,19 +185,38 @@ func attachVolumes(ctx *context, ops map[params.MachineStorageId]*attachVolumeOp
 				}
 				reschedule = append(reschedule, ops[id])
 
+				resultErr := result.Error
+				if resultErr != nil {
+					entityStatus.Info = resultErr.Error()
+				}
+
+				var respErr *azcore.ResponseError
```

Comment:
Don't do this here. The worker needs to be provider agnostic. 
The details of interpreting the error and surfacing an messages are the responsibility of the provider.

Reply:
Will set err to the body for this case instead, thanks ian!